stan/math/prim/functor/apply_vector_unary.hpp

```diff
@@ -17,6 +17,52 @@ namespace math {
 template <typename T, typename Enable = void>
 struct apply_vector_unary {};
 
+/**
+ * Base template class for vectorization of unary vector functions
+ * defined by applying a functor to a standard library vector, Eigen dense
+ * matrix expression template, or container of these. For each specialisation,
+ * the same vector type as the input is returned.
+ *
+ * Two taxonomies of unary vector functions are implemented:
+ * - f(vector) -> vector
+ * - f(vector) -> scalar
+ *
+ * This base template class takes (and returns) Eigen expression templates.
+ */
+template <typename T>
+struct apply_vector_unary<T, require_var_matrix_t<T>> {
+  /**
+   * Member function for applying a functor to a vector and subsequently
+   * returning a vector.
+   *
+   * @tparam T Type of argument to which functor is applied.
+   * @tparam F Type of functor to apply.
+   * @param x Eigen input to which operation is applied.
+   * @param f functor to apply to Eigen input.
+   * @return Eigen object with result of applying functor to input
+   */
+  template <typename F>
+  static inline T apply(const T& x, const F& f) {
```

Comment:
The other versions specialize on whether or not `T` contains something that is array-like or matrix-like and do an `eval`.

I do not think we return expression-like things from `var_value` functions -- they're always `var_value` s, and the way to make sure the input and output types match is to just put `T` in both places right?